### DISCUSSION TITLE
Is the skeleton different from the original one?

### DISCUSSION THREAD

**Discussion by emrehan98** [Sixze/ALS-Refactored#39] [Q&A] [answered]
Today I was importing some of the old animations I had and I realized they look different from the blueprint version of the ALS. Here's the screenshot:

![image](https://user-images.githubusercontent.com/19621279/140614176-9687a754-0861-483b-b027-74fc6852c294.png)

I tested in fresh project with ALS Refactored and fresh project in original ALS. Results were the same.

If you made any changes could you give some tips about what to do? I plan to replace current skeleton with the old one but if you have other workaround I would love to try that first.

**Comment by Sixze** [comment]
Hmmm, for the skeleton I only modified the virtual bones, sockets, retargeting options and blend profiles. It looks like a problem with retargeting, so try changе retargeting options, it should help.

**Reply by Sixze** [reply]
Or maybe it's just an UE problem. Try restart the editor after importing the animation.

**Reply by emrehan98** [reply]
Thanks for answering! I've played with import settings but had no luck. I both tried to retargeting and importing from .fbx file which was exported from Blender. Results are different on ALS Refactored version for some reason.

Not asking for you to try or anything but if you are interested, here's the animation file that you can quickly add to your project. I'll try to do more tests. I think about removing the skeleton and trying to use original one but that would create tons of work. Still need to do more tests.

[RopeClimbing_Swing_Backwards.zip](https://github.com/Sixze/ALS-Refactored/files/7491080/RopeClimbing_Swing_Backwards.zip)

Edit: Just realized the reference transform and the transform of the bone is not matching. If I change the transform and make it identical to reference transform, it looks normal. But I don't know what causes to have different transforms in the first place.

**Reply by Sixze** [reply]
Thanks for the animation, I'll see what's wrong and if I can figure it out I'll fix the skeleton.

**Comment by Sixze** [comment, marked as answer]
The animation was made for a mesh with proportions different from the proportions of the mannequin from ALS, so for this case you need to set up retargeting in a special way:
1. Add the mesh for which the animation was originally made to the list of retargeting sources in the skeleton asset.
>
![image](https://user-images.githubusercontent.com/16257871/140644360-ce14e827-b9de-4b63-a61f-3e2457295a8b.png)
2. In the animation asset, select the correct retarget source, after that everything should work as expected.
>
![image](https://user-images.githubusercontent.com/16257871/140644450-1af71fd3-9aae-4764-bdf7-ba3c610a94d3.png)

**Reply by Sixze** [reply]
There is another way to fix this problem, but the solution that I suggested is probably the most correct.

**Comment by emrehan98** [comment]
Thank you so much for taking your time and solving the problem Sixze, I wouldn't do it by myself. It works like a charm!

**Comment by emrehan98** [comment]
I noticed some problem after applying the solution. Asset looks great when the retargeting is applied but when you put it into animBP, artifacts start appearing. (In one of my animation, the sholder of the character was getting bigger and disoriented.) So after your solution, we just need to do couple more things.

![image](https://user-images.githubusercontent.com/19621279/140647463-cdb2d66d-00ec-415b-adf4-69eb27763032.png)
Basically I do everything you did and after that, I created a new asset with the method I showed above. Then I just started using that animation, instead of using the retargeted one.

I also created a fresh project in order to keep things clean. So every broken animation is being fixed with this method and then imported to the base project.

**Reply by Sixze** [reply]
Didn't know about this feature, seems to be very useful.

You can also try the second method of retargeting, just open the skeleton asset, right-click on the pelvis bone and select "Skeleton", maybe this method will work better than the first one I suggested.

![image](https://user-images.githubusercontent.com/16257871/140813081-e0676eb9-4764-4a6e-8510-fbbf3213c4ed.png)
